src/executors/mod.rs

```diff
@@ -19,12 +21,30 @@ fn get_function_output<'a>(
     let handler = function.handler.as_ref(py);
     let request_hashmap = request.to_hashmap(py).unwrap();
 
-    // this makes the request object accessible across every route
-    match function.number_of_params {
+    if function.validate_params {
+        let body = request.body.clone().to_vec();
+        let json_str = str::from_utf8(&body).unwrap();
+
+        // Use Python to convert back to PyDict after deserializing
+        let json = py.import("json").unwrap();
+
+        // Extract the route's dependencies from the FunctionInfo
+        let json_dict = json.call_method1("loads", (json_str,)).unwrap();
```

Comment:
I suppose you were also unable to find a crate that would allow us to convert a string to a hashmap or a pydict. I think we will have to write a `macro!` for this. I was afraid things might come to this. Maybe Pydantic v2 could help us here?

Reply:
We can leave it in if you're not very familiar with macros! I'm not very either. I'll have to read more of it.

Reply:
I'll read more of it too!

Reply:
I've just realized that pythonize (by the author of PyO3) exists. If we convert the string to a serde_json::Value, we can use   pythonize to convert the Value enum/object into a PyDict (since Value implements the Deserialize trait). I am going to try this approach.

Reply:
Perfect @kliwongan 😄 Let me know if you need any help from my end 😄

Reply:
@sansyrox Update, I found out that msgspec's schema validation is very nice, so I am using that to pass in the query parameters (instead of Pythonizing, I get the output of the model decode/schema validation in the form of the objects/default classes for the function signature) however I have tested the Pythonize approach and that works. I am also thinking of moving the query param validation in get_function_output to put all of the robyn Rust imports in one place. This, and making sure ForwardRefs work properly will take time, but I will complete this in time.